salt/modules/swarm.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+"""
+Docker Swarm Module using Docker's Python SDK
+=============================================
+
+:codeauthor: Tyler Jones <jonestyler806@gmail.com>
+
+.. versionadded:: 2017.7.2
+
+The Docker Swarm Module is used to manage and create Docker Swarms.
+
+Dependencies
+============
+
+- Docker installed on the host
+- Docker python sdk >= 2.5.1
+
+Docker Python SDK
+=================
+pip install -U docker
+
+More information: https://docker-py.readthedocs.io/en/stable/
+"""
+from __future__ import absolute_import
+import docker
```

Comment:
This needs to be 

```
try:
    import docker
    HAS_DOCKER = True
except ImportError
    HAS_DOCKER = False
```

and then a `__virtual__` function that explains why the module wasn't loaded.

Looking at the example in this blog post should help.

https://blog.gtmanfred.com/contributing-to-salt-part-1.html

https://docs.saltstack.com/en/latest/ref/modules/#virtual-modules